python/valis/routes/query.py

```diff
@@ -85,3 +85,19 @@ async def cone_search(self,
         """ Perform a cone search """
         return list(cone_search(ra, dec, radius, units=units))
 
+    @router.get('/carton_program_all', summary='Return a list of all carton or programs',
+                response_model=list, dependencies=[Depends(get_pw_db)])
+    async def carton_program_all(self,
+                                 name_type: str = Query('carton', enum=['carton', 'program'],
+                                                        description='Specify search on carton or program', example='carton')):
+        """ Return a list of all carton or programs """
+        return carton_program_list(name_type)
+
+    @router.get('/carton_program', summary='Search for all SDSS targets within a carton or program',
```

Comment:
Same thing with the name here.  Will this be confusing for the user that it's searching for targets by carton/program rather than directly for cartons/programs?  This is consistent with the `/cone` endpoint though.

Reply:
Ok, I did add a hyphen to the name to make it more human readable (https://github.com/sdss/valis/pull/6/commits/a88f41e2ded592f9b13d3714695481ce00701a85).  I do like keeping them in the same endpoint function though? I agree it operates very similar to `/cone`, so I don't think it would be too confusing. I can separate it into two endpoints though if you think that would be easier to deal with on the front end.

Reply:
I also like that the search route is one route.  That will be easier for the front-end I think.  I would just update the endpoint name to `/cartons-progams`.

Reply:
actually we can leave it as singular here.

Reply:
Ok, sounds good!